ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/vlan-interfaces.j2

```diff
@@ -1,156 +1,166 @@
    ipv6 access-group {{ vlan_interfaces[vlan_interface].ipv6_access_group_in }} in
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ipv6_access_group_out is defined and vlan_interfaces[vlan_interface].ipv6_access_group_out is not none %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ipv6_access_group_out is arista.avd.defined %}
    ipv6 access-group {{ vlan_interfaces[vlan_interface].ipv6_access_group_out }} out
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_network_point_to_point is defined and vlan_interfaces[vlan_interface].ospf_network_point_to_point == true %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_network_point_to_point is arista.avd.defined(true) %}
    ip ospf network point-to-point
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_area is defined and vlan_interfaces[vlan_interface].ospf_area is not none %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_area is arista.avd.defined %}
    ip ospf area {{ vlan_interfaces[vlan_interface].ospf_area }}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_cost is defined and vlan_interfaces[vlan_interface].ospf_cost is not none %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_cost is arista.avd.defined %}
    ip ospf cost {{ vlan_interfaces[vlan_interface].ospf_cost }}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_authentication is defined and vlan_interfaces[vlan_interface].ospf_authentication is not none %}
-{%           if vlan_interfaces[vlan_interface].ospf_authentication == "simple" %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_authentication is arista.avd.defined %}
+{%       if vlan_interfaces[vlan_interface].ospf_authentication == "simple" %}
    ip ospf authentication
-{%           elif vlan_interfaces[vlan_interface].ospf_authentication == "message-digest" %}
+{%       elif vlan_interfaces[vlan_interface].ospf_authentication == "message-digest" %}
    ip ospf authentication message-digest
-{%           endif %}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_authentication_key is defined and vlan_interfaces[vlan_interface].ospf_authentication_key is not none %}
+{%       endif %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_authentication_key is arista.avd.defined %}
    ip ospf authentication-key 7 {{ vlan_interfaces[vlan_interface].ospf_authentication_key }}
-{%         endif %}
-{%         if vlan_interfaces[vlan_interface].ospf_message_digest_keys is defined and vlan_interfaces[vlan_interface].ospf_message_digest_keys is not none %}
-{%           for ospf_message_digest_key in vlan_interfaces[vlan_interface].ospf_message_digest_keys | arista.avd.natural_sort %}
+{%     endif %}
+{%     if vlan_interfaces[vlan_interface].ospf_message_digest_keys is arista.avd.defined %}
```

Comment:
this check is included in arista.avd.natural_sort, so you can remove this.

Reply:
Fixed with  07d5adb6